Fix a bug in feff.inputs

kaifengZheng opened the pull request:
## Summary

Major changes:

- features: fix the bug in the feff.inputs

- fix 1: .delete  atom_sym = get_absorbing_atom_symbol_index(absorbing_atom, self._cluster)[0]
- fix 2:   change the argument: self.pot_dict = get_atom_map(self._cluster, self.absorbing_atom)

## Todos

If this is work in progress, what else needs to be done?

- feature 2: ...
- fix 2:

## Checklist

- [ ] Google format doc strings added. Check with `ruff`.
- [ ] Type annotations included. Check with `mypy`.
- [x] Tests added for new features/fixes.
- [ ] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

kaifengZheng commented:
Fix bug in feff.inputs

janosh commented:
What's the bug you encountered?

kaifengZheng commented:
Hi Janosh,
I found a small bug inside the feff.inputs module.

atom_sym = get_absorbing_atom_symbol_index(absorbing_atom, self._cluster)[0]

The argument 'absorbing_atom' is misused, which generated bugs. Based on my understanding, this "absorbing_atom" is the index of the absorbing atom in the original structure, but this sentence misused this index for the absorbing atom in the generated cluster, which should be 0.

Firstly, I think I need to change this argument to 0. However, after I scrutinize the codes, I found that this sentence is not useful, so I delete it.

Please let me know what you think.

Best,
Kaifeng

janosh commented:
Could you post a minimal example that demonstrates the bug? We also need a test that failed before and passes with this change.

kaifengZheng commented:
Dear Janosh,
I prepared two notebooks(in the zip folder) to describe this problem. One notebook named test_beforechange.ipynb is the test before I modified the code, and there is an out-of-range problem. Another notebook shows how the program ran well after I modified the code, and I mentioned two ways to change the code. Either approach should work.

By the way, I need to learn how to pass the workflow test. If the first modification approach doesn't work, I will try the second one.

Best regards,
Kaifeng Zheng


[pymatgen_test.zip](https://github.com/materialsproject/pymatgen/files/12340230/pymatgen_test.zip)

kaifengZheng commented:
Sorry, I forgot to send the requirements.txt for packages

[requirements.txt](https://github.com/materialsproject/pymatgen/files/12340296/requirements.txt)

kaifengZheng commented:
@janosh Hi Janosh, how's thing going? Please let me know if I need to provide more support.

kaifengZheng commented:
Hi @janosh, do I need to provide more information?

janosh commented:
Yes, as I mentioned above, we're missing a new test for this fix.

kaifengZheng commented:
Hi @janosh, I sent a zip file containing the test. The html files also show the running results before and after the changes. If something else is needed, please let me know. What is the format for the test file? Do I need to create a folder and upload it via github?

janosh commented:
Please have a look at the existing tests in https://github.com/materialsproject/pymatgen/blob/0f352bfd6050a4ad362c6ff82ed5123dde36fed6/tests/io/feff/test_sets.py#L17

kaifengZheng commented:
Hi @janosh, I just updated my test files. I appended a function in test_sets.py named test_cluster_index(), which will check the functionalities of cluster creation.
The old code will create the error(IndexError):
![image](https://github.com/materialsproject/pymatgen/assets/48105165/c5b76286-42b4-43f8-b9f8-926ed8e25da6)

The index problem can also lead to a mis-indexing of the absorbing atom for simple crystals when is running well without exception, However, I think it is fine since the absorber atom is deleted from the potential list in the 'get_atom_map' function, 

This is my first time to contribute on GitHub. Please let me know if I need to provide more information. Thanks
